RELEASE/scripts/autoscend/auto_providers.ash

```diff
@@ -1212,3 +1212,252 @@ boolean provideMoxie(int amt, boolean doEquips)
+	if(doEverything && pathHasFamiliar() && pathAllowsChangingFamiliar())
+	{
+		if(!speculative)
+		{
+			handleFamiliar("meat");
+		}
+		// fam isn't equipped immediatly even if we aren't speculating
+		// so add bonus from fam regardless of speculation
+		familiar target = lookupFamiliarDatafile("meat");
+		if(target != $familiar[none] && target != my_familiar())
+		{
+			int famWeight = familiar_weight(target) + weight_adjustment();
+			delta += numeric_modifier(target, "Meat Drop",famWeight,$item[none]);
+			auto_log_debug("With using familiar: " + target + " we can get to " + result());
+		}
+		else
+		{
+			auto_log_debug("Already have desired familar, " + target + ", active.");
+		}
+		if(pass())
+			return result();
+	}
+	void handleEffect(effect eff)
+	{
+		if(speculative)
+		{
+			delta += numeric_modifier(eff, "Meat Drop");
+		}
+		auto_log_debug("We " + (speculative ? "can gain" : "just gained") + " " + eff.to_string() + ", now we have " + result());
+	}
+	boolean tryEffects(boolean [effect] effects)
+	{
+		foreach eff in effects
+		{
+			if(buffMaintain(eff, 0, 1, 1, speculative))
+				handleEffect(eff);
+			if(pass())
+				return true;
+		}
+		return false;
+	}
+	if((get_property("sidequestArenaCompleted") == "fratboy") && !get_property("concertVisited").to_boolean() && (have_effect($effect[Winklered]) == 0))
+	{
+		if(is_professor())
+		{
+			//Need to manually equip because professor
+			if(!have_equipped($item[beer helmet])) equip($item[beer helmet]);
+			if(!have_equipped($item[distressed denim pants])) equip($item[distressed denim pants]);
+			if(!have_equipped($item[bejeweled pledge pin])) equip($item[bejeweled pledge pin]);
+		}
+		else
+		{
+			outfit("Frat Warrior Fatigues");
+		}
+		cli_execute("concert 2");
+	}
+	handleBjornify($familiar[Hobo Monkey]);
+	// unlimited skills
+	if(tryEffects($effects[
+		Polka of Plenty,
+		Disco Leer
+	]))
+		return result();
+	if(canAsdonBuff($effect[Driving Observantly]))
+	{
+		if(!speculative)
+			asdonBuff($effect[Driving Observantly]);
+		handleEffect($effect[Driving Observantly]);
+	}
+	if(pass())
+		return result();
+	if(bat_formWolf(speculative))
+	{
+		handleEffect($effect[Wolf Form]);
+	}
+	if(pass())
+		return result();
+	if(auto_birdModifier("Meat Drop") > 0)
+	{
+		if(tryEffects($effects[Blessing of the Bird]))
+			return result();
+	}
+	if(auto_favoriteBirdModifier("Meat Drop") > 0)
+	{
+		if(tryEffects($effects[Blessing of Your Favorite Bird]))
+			return result();
+	}
+	if(isActuallyEd())
+	{
+		if(!have_skill($skill[Gift of the Maid]) && ($servant[Maid].experience >= 441))
+		{
+			visit_url("charsheet.php");
+			if(have_skill($skill[Gift of the Maid]))
+			{
+				auto_log_warning("Gift of the Maid not properly detected until charsheet refresh.", "red");
+			}
+		}
+		if(tryEffects($effects[
+		Purr of the Feline
+		]))
+		return result();
+	}
+	songboomSetting("meat");
+	// items
+	if(tryEffects($effects[
+		Greedy Resolve,
+		Kindly Resolve,

```

Comment:
the +fam buffs belong since we don't have a fam weight provider yet. To help implement that in the future, please add comments to buffs which are only for fam weight. Makes it easier to see which ones can get pulled out later

Reply:
One of the things I had wanted to do was say what each buff did so this motivates me to do that